src/ReactiveUI/ReactiveCommand/ReactiveCommand.cs

```diff
@@ -689,10 +688,11 @@ public class ReactiveCommand<TParam, TResult> : ReactiveCommandBase<TParam, TRes
                 }).StartWith(false).CombineLatest(_isExecuting, (canEx, isEx) => canEx && !isEx)
                 .DistinctUntilChanged()
                 .Replay(1).RefCount();
+
             _results = _synchronizedExecutionInfo.Where(x => x.Demarcation == ExecutionDemarcation.Result)
                 .Select(x => x.Result);
 
-            _canExecuteSubscription = _canExecute.Subscribe(OnCanExecuteChanged);
+            _canExecuteSubscription = _canExecute.ObserveOn(RxApp.MainThreadScheduler).Subscribe(OnCanExecuteChanged);
         }
```

Comment:
This needs to be configurable and not the default due to performance.

Reply:
added canExecuteScheduler parameter defaulted to RxApp.MainThreadScheduler